web/html/src/components/input/Form.js

```diff
@@ -45,40 +44,52 @@ class Form extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props);
     this.state = {
-      isValid: true,
+      fieldsStatus: []
     };
   }
 
   setModelValue(name: string, value: any) {
     const { model } = this.props;
     if (value == null && model[name] != null) {
       delete model[name];
-      const isValid = this.validateForm(model);
-      this.props.onChange(model, isValid);
+      this.validateForm(model);
+      this.props.onChange(model);
     } else if (value != null) {
       model[name] = value;
-      const isValid = this.validateForm(model);
-      this.props.onChange(model, isValid);
+      this.validateForm(model);
+      this.props.onChange(model);
     }
   }
 
-  validateForm(model: Object): boolean {
-    let allValid = true;
+  validateForm(model: Object) {
+    const fieldsStatus = Object.keys(this.inputs).reduce((ret, name) => {
+      ret[name] = this.inputs[name].validate(model);
+      return ret;
+    }, {});
 
-    Object.keys(this.inputs).forEach((name) => {
-      if (!this.inputs[name].validate(model)) {
-        allValid = false;
-      }
-    });
+    const allValid = Object.values(fieldsStatus).every(status => status);
```

Comment:
Even though it's a one-liner, let's put this in a local method called `isAllValid()` since it's used multiple times. You can also directly assign the function call without the need for the `allValid` var.